lang/java/reef-common/src/main/java/org/apache/reef/runtime/common/driver/evaluator/EvaluatorDescriptorImpl.java

```diff
@@ -36,17 +38,28 @@
   private final int numberOfCores;
   private EvaluatorProcess process;
   private final String runtimeName;
+  private final Set<String> nodeLabelExpressions;
```

Comment:
I wanted to use empty optional object instead of null in the constructor but It doesn't work as I expected. I will dig it up.